### PR TITLE
3239/moved shared helper components

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue Overview

This PR addresses #3239

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This relocates several components from ui-components to shared-helpers in this repo as described in the issue AC. In addition, the structure of the files in shared-helpers has been improved to delineate between auth helpers, utility functions, and various view templates.

Note that a subsequent ui-components PR will be needed to remove the components which were added here.

## How Can This Be Tested/Reviewed?

Various parts of the UI in public and partners function as expected with no changes in behavior.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/src/pages/sign-in.tsx [hunk 1]
```diff
@@ -137,7 +133,7 @@ const SignIn = () => {
           onSubmit={onSubmit}
           control={{ register, errors, handleSubmit, watch }}
           networkStatus={{
-            content: { ...networkStatusContent, error: !!networkStatusContent?.error },
```
Comment (emilyjablonski) [inline sites/public/src/pages/sign-in.tsx:140]: Curious what prompted these changes?

**Comment by emilyjablonski** [comment]
I am loving the new organization! 🥳

File: shared-helpers/src/views/multiselectQuestions.tsx [hunk 2]
```diff
@@ -275,10 +276,12 @@ export const getCheckboxOption = (
 }
 
 export const mapRadiosToApi = (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   data: { [name: string]: any },
   question: MultiselectQuestion
 ): ApplicationMultiselectQuestion => {
   const [key, value] = Object.entries(data)[0]
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
[the remark this answers is not in the document]
Comment (jaredcwhite) [reply shared-helpers/src/views/multiselectQuestions.tsx]: @ludtkemorgan I was able to update a couple of those spots, but a couple of other ones didn't seem easy to change or work around.

**Review by ludtkemorgan — APPROVED** [review]
Thanks for making those changes

**Review by emilyjablonski — APPROVED** [review]

**Pull request merged**